core/storage/base_model/gae_models_test.py

```diff
@@ -235,3 +348,24 @@ def test_get_multi_versions_errors(self):
             ValueError,
             'At least one version number is invalid'):
             TestVersionedModel.get_multi_versions('model_id1', [1, 1.5, 2])
+
+
+class TestBaseModel(base_models.BaseModel):
+    """Model that inherits BaseModel for testing. This is required as BaseModel
+    gets subclassed a lot in other tests and that can create unexpected errors.
+    """
+    pass
+
+
+class BaseModelTests(test_utils.GenericTestBase):
+
+    def test_raise_exception_by_mocking_collision(self):
+        with self.assertRaisesRegexp(
+            Exception, 'New id generator is producing too many collisions.'):
+            # Swap dependent method get_by_id to simulate collision every time.
+            with self.swap(
```

Comment:
This is very hard to read, please refactor the context managers using temporary variables.

Reply:
Done